### PR TITLE
DRACO-3588 added events- get reviewed part-1

### PR DESCRIPTION
**Pull request by drupalsyed**

### CODE REVIEW

**Comment by drupalsyed** [comment]
This PR supports conditional publishing to youtube.
Testing steps:-
- Add a media type and configure it accordingly to upload local videos.
- Add a media to the using the media type created.
- Select publishing option as per your need, Else the default action will be performed.

File: src/EventSubscriber/YoutubeEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  /**
+   * The event handler to publish a media.
+   *
+   * @param \Drupal\yt_push\Event\YoutubeMediaEvent $event
+   *   An YoutubeMediaEvent event.
+   */
+  public function doPublishMedia(YoutubeMediaEvent $event) {
+    $media = $event->getMedia();
+    if ($media instanceof MediaInterface && $event->checkPublish()) {
+      $this->youTubeUploader->uploadLocalFile($media);
```
Comment (m4olivei) [inline src/EventSubscriber/YoutubeEventSubscriber.php]: Hum, if you upload the video from this event subscriber, how would another event subscriber prevent the video from being uploaded to YouTube?

File: src/Event/YoutubeMediaEvent.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  /**
+   * Check Publish.
+   *
+   * @return bool
+   *   Publish Condition.
+   */
+  public function checkPublish() {
+    return $this->doPublish === '' ? $this->mediaEntity->isPublished() : $this->doPublish;
```
Comment (m4olivei) [inline src/Event/YoutubeMediaEvent.php]: Is this the best place for checking the published checkbox on the media entity? This hardcodes the logic of our default conditional check into the event. How would a brand implement their own logic given that this is hardcoded here?

File: src/EventSubscriber/YoutubeEventSubscriber.php (hunk 1 above)
Comment (m4olivei) [inline src/EventSubscriber/YoutubeEventSubscriber.php]: Hum, if you upload the video from this event subscriber, how would another event subscriber prevent the video from being uploaded to YouTube?

**Comment by m4olivei** [comment]
@drupalsyed can you fix the merge conflict here?

File: src/EventSubscriber/YoutubePublishEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
```
Comment (m4olivei) [inline src/EventSubscriber/YoutubePublishEventSubscriber.php]: Note that this file should be named `YouTubePublishEventSubscriber` (capital "T"). As it is it is causing errors on the Behat test job.

**Pull request merged**